src/main/java/com/google/jenkins/plugins/computeengine/ComputeEngineInstance.java

```diff
@@ -23,37 +23,41 @@
 import hudson.slaves.AbstractCloudComputer;
 import hudson.slaves.AbstractCloudSlave;
 import hudson.slaves.ComputerLauncher;
-import hudson.slaves.NodeProperty;
 import hudson.slaves.RetentionStrategy;
 import java.io.IOException;
 import java.util.Collections;
-import java.util.Optional;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 import jenkins.model.Jenkins;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Setter;
 
+@Getter
+@Setter
 public class ComputeEngineInstance extends AbstractCloudSlave {
   private static final long serialVersionUID = 1;
   private static final Logger LOGGER = Logger.getLogger(ComputeEngineInstance.class.getName());
 
   // TODO: https://issues.jenkins-ci.org/browse/JENKINS-55518
-  public final String zone;
-  public final String cloudName;
-  public final String sshUser;
-  public final transient Optional<WindowsConfiguration> windowsConfig;
-  public final boolean createSnapshot;
-  public final boolean oneShot;
-  public Integer launchTimeout; // Seconds
+  private String zone;
+  private String cloudName;
+  private String sshUser;
+  private transient WindowsConfiguration windowsConfig;
+  private boolean createSnapshot;
+  private boolean oneShot;
+  private Integer launchTimeout; // Seconds
   private Boolean connected;
 
-  public ComputeEngineInstance(
+  @Builder
+  private ComputeEngineInstance(
       String cloudName,
       String name,
       String zone,
       String nodeDescription,
       String sshUser,
       String remoteFS,
-      Optional<WindowsConfiguration> windowsConfig,
+      WindowsConfiguration windowsConfig,
```

Comment:
What's the thought behind removing the Optional wrapper?

Reply:
Mainly for consistency with InstanceConfiguration, which requires this to be non-optional for configuration as code, as Optional is not serializable. Here, it's not as important, and there is flexibility from being able to put @Builder on the constructor instead of the class. I would still opt to avoid optional for the builder/constructor parameter but leave the field type as Optional so that the getter will return the optional value.